Sources/swift-bundler/Commands/BundleCommand.swift

```diff
@@ -79,6 +79,15 @@ struct BundleCommand: AsyncCommand {
         return false
       }
 
+      if platform == .visionOS,
+        builtWithXcode || arguments.universal || !arguments.architectures.isEmpty
+      {
+        log.error(
+          "'--built-with-xcode', '--universal' and '--arch' are not compatible with '--platform visionOS'"
+        )
+        return false
+      }
```

Comment:
Try combining this with the previous if-statement (for iOS) to reuse the code. The only thing that changes between the two is the platform name in the error message.